### PR TITLE
ovmfvartool: use as a python package 

### PR DESCRIPTION
**Pull request by baloo**
###### Description of changes

Ovmfvartool now comes as a standard python package. We (@RaitoBezarius and I) intend to consume the API directly in the test-driver for NixOS tests to interact with OVMF variables.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]
LGTM.

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 1]
```diff
@@ -1,28 +1,17 @@
-{ lib, stdenvNoCC, fetchFromGitHub, python3 }:
+{ lib, buildPythonPackage, fetchFromGitHub, pyyaml, fetchpatch }:
```
Comment (marsam) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:1]: fetchpatch is unused
```suggestion
{ lib, buildPythonPackage, fetchFromGitHub, pyyaml }:
```

File: pkgs/applications/virtualization/ovmfvartool/default.nix [hunk 2]
```diff
@@ -1,28 +1,17 @@
...
-    # needed in build but /usr/bin/env is not available in sandbox
-    substituteInPlace ovmfvartool \
-      --replace "/usr/bin/env python3" "${pythonPkg.interpreter}"
-  '';
-
-  installPhase = ''
-    mkdir -p $out/bin
-    install -m 755 ovmfvartool $out/bin/
-  '';
+  propagatedBuildInputs = [ pyyaml ];
```
Comment (marsam) [inline pkgs/applications/virtualization/ovmfvartool/default.nix:15]:
```suggestion
  # has no tests
  doCheck = false;

  pythonImportsCheck = [ "ovmfvartool" ];

```

**Review by marsam — COMMENTED** [review]
would you mind changing the `license`  to `licenses.gpl3Plus`?

**Comment by baloo** [comment]
@marsam thanks for the review!

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -23280,8 +23280,6 @@ with pkgs;
     tpmSupport = true;
   };
 
-  ovmfvartool = callPackage ../applications/virtualization/ovmfvartool { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:23283]: We probably want to keep this, to not break some people using it as an application

```
  ovmfvartool = with python3.pkgs; toPythonApplication ovmfvartool;
```

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -6452,6 +6452,8 @@ in {
 
   ovh = callPackage ../development/python-modules/ovh { };
 
+  ovmfvartool = callPackage ../applications/virtualization/ovmfvartool { };
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:6455]: if it's going to be listed in python-packages.nix, then we should probably also move the file to `python-modules`
```suggestion
  ovmfvartool = callPackage ../development/python-modules/ovmfvartool { };
```

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (baloo) [reply pkgs/top-level/all-packages.nix:23283]: it was included only a few days ago. I don't think this is too bad. I think I'm only consumer for now :D

**Review by marsam — APPROVED** [review]
LGTM

**Pull request merged**